### PR TITLE
Migrate hf trainer

### PR DESCRIPTION
**Pull request by gkumbhat**
### Changes

- Replace custom training loop with HF Trainer

### CODE REVIEW

**Review by Ssukriti — CHANGES_REQUESTED** [review]
from the changes it seems like fine tuning was already using Trainer, you have now added it to prompt tuning as well. I think you meant to re-use some functions, hence you moved them to toolkit.
Prompt tuning is using `launch_training` and preprocessing from toolkit, but text_generation_local has its own definition of both functions, `_launch_training`. Code looks the same between both launch training, so maybe you forgot to change fine tuning to use toolkit as well?

**Comment by gkumbhat** [comment]
@Ssukriti , I did this on purpose to avoid making too many refactors in one PR.  I added a comment regarding that: https://github.com/caikit/caikit-nlp/pull/287/files#diff-3ca8e28141febc0ff5a812a8b7a2f92997f0098406b924bd4bdbcab680813cc3R73

File: caikit_nlp/modules/text_generation/peft_prompt_tuning.py [hunk 1]
```diff
@@ -356,18 +355,47 @@ def train(
             "<NLP46653367E>", len(train_stream) > 0, "train_stream cannot be empty"
         )
 
+        if accumulate_steps:
```
[the remark this answers is not in the document]
Comment (Ssukriti) [reply caikit_nlp/modules/text_generation/peft_prompt_tuning.py:358]: the experiments only capture train time. accumulation steps has known to influence accuracy (we have experiments proving it ) . there is a tradeoff between train time and accuracy. I agree larger values of 16 can drastically slow down training. we are leaning towards a default of 4 as it gives best accuracy. I can re-run experiments with acc=4 to measure train time.

in any case, how about instead of deprecating hyperparameters in library, we make a guideline that informs what users pick and give them the choice to tune it based on the guideline?

**Review by Ssukriti — APPROVED** [review]
I have tested PT before and after the change with llama model on some quick text samples, and results are same.
